src/main/java/org/sonarsource/sonarlint/ls/EnginesFactory.java

```diff
@@ -102,22 +104,29 @@ StandaloneSonarLintEngine newStandaloneEngine(StandaloneGlobalConfiguration conf
   }
 
   public ConnectedSonarLintEngine createConnectedEngine(String connectionId) {
-    ConnectedGlobalConfiguration configuration = ConnectedGlobalConfiguration.builder()
+    ConnectedGlobalConfiguration.Builder builder = ConnectedGlobalConfiguration.builder()
       .setConnectionId(connectionId)
       .setExtraProperties(prepareExtraProps())
       .addEnabledLanguages(STANDALONE_LANGUAGES)
       .addEnabledLanguages(CONNECTED_ADDITIONAL_LANGUAGES)
       .setNodeJs(nodeJsRuntime.getNodeJsPath(), nodeJsRuntime.getNodeJsVersion())
       .setModulesProvider(modulesProvider)
-      .setLogOutput(lsLogOutput)
-      .build();
+      .setLogOutput(lsLogOutput);
 
-    ConnectedSonarLintEngine engine = newConnectedEngine(configuration);
+    extraAnalyzers.forEach(analyzer-> builder.addExtraPlugin(resolvePluginKey(analyzer.getPath()), analyzer));
+    ConnectedSonarLintEngine engine = newConnectedEngine(builder.build());
 
     LOG.debug("SonarLint engine started for connection '{}'", connectionId);
     return engine;
   }
 
+  static String resolvePluginKey(String pluginUrl) {
```

Comment:
Maybe a better name would be:
```suggestion
  static String guessPluginKey(String pluginUrl) {
```
?